optuna/artifact/_protocol.py

```diff
@@ -0,0 +1,59 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+
+try:
+    from typing import Protocol
+except ImportError:
+    from typing_extensions import Protocol  # type: ignore
+
+
+if TYPE_CHECKING:
+    from typing import BinaryIO
+
+
+class ArtifactStore(Protocol):
+    """A protocol defining the interface for an artifact backend.
+
+    An artifact backend is responsible for managing the storage and retrieval
+    of artifact data. The backend should provide methods for opening, writing
+    and removing artifacts.
+    """
+
+    def open(self, artifact_id: str) -> BinaryIO:
```

Comment:
I understand you took the implementation from Optuna Dashboard, which is added by me. But the name `open()` is a bit confusing since it looks as if it can open a file in write mode.

`get_reader()` or `open_reader()` might be more specific and clearer. @contramundum53 @gen740 What do you think?

Reply:
I agree.

I favor the name `open_reader()` as it's more natural when used within a `with` statement's scope.

Reply:
Sounds reasonable 👍